.github/workflows/build.yml

```diff
@@ -14,21 +14,26 @@ jobs:
   build:
     name: Maven build
     runs-on: ubuntu-18.04
-
     strategy:
       matrix:
         maven-version:
           - '3.3.9'
           - '3.6.1'
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           fetch-depth: 2
       - name: Set up JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: 'zulu'
+      - uses: actions/cache@v2
```

Comment:
Is there a reason for using the separate `actions/cache` action instead of the [Maven caching support](https://github.com/actions/setup-java#caching-maven-dependencies) added in  [actions/setup-java@v2.3.0](https://github.com/actions/setup-java/releases)?
Regardless, this is an unusually project in that almost every change to this repo is a `pom.xml` change (e.g. dependency updates from Dependabot). The vast majority of code changes happen upstream in `ktlint` itself. So, enabling caching in this project will have a net effect of slowing down the builds. Almost every build will have to spend time tarring and uploading the Maven repo, just to discard it for the next build as the `pom.xml` has changed.

Reply:
5ee89ce26e2359a1abd6cdd4069c31baba186dfd